app/src/androidTest/java/com/github/h3lp3rs/h3lp/forum/ForumCacheTest.kt

```diff
@@ -0,0 +1,166 @@
+package com.github.h3lp3rs.h3lp.forum
+
+import androidx.test.core.app.ApplicationProvider
+import com.github.h3lp3rs.h3lp.database.Databases.*
+import com.github.h3lp3rs.h3lp.database.Databases.Companion.setDatabase
+import com.github.h3lp3rs.h3lp.database.MockDatabase
+import com.github.h3lp3rs.h3lp.forum.implementation.CachedForum
+import com.github.h3lp3rs.h3lp.signin.SignInActivity.Companion.globalContext
+import com.github.h3lp3rs.h3lp.storage.Storages.Companion.resetStorage
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+class CacheForumTest {
+
+    // Useful variables
+    private lateinit var rawForum: Forum
+    private lateinit var cachedForum: Forum
+
+    @Before
+    fun setup() {
+        globalContext = ApplicationProvider.getApplicationContext()
+        resetStorage()
+        setDatabase(PREFERENCES, MockDatabase())
+        ForumCategory.mockForum()
+        rawForum = ForumCategory.root().child(ForumCategory.GENERAL.name)
+        cachedForum = CachedForum(rawForum)
+    }
+
+    private fun resetOnlineForum() {
+        ForumCategory.mockForum()
+        rawForum = ForumCategory.root().child(ForumCategory.GENERAL.name)
+        cachedForum = CachedForum(rawForum)
+    }
+

```

Comment:
As you can see in the comments in the CachedForum, if you tested a user posting on the cached forum, then getting the post they posted, they wouldn't actually see it (for the reasons mentioned below).

Reply:
As discussed, this is not the case